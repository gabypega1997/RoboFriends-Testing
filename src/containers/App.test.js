import React from "react";

import { shallow, mount } from "enzyme/build";
import App from "./App"
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer'
import { Provider } from "react-redux";


const mockStore = configureStore([]);

describe('My Conection React-Redux',()=>{
    let store;
    let component;
    
    beforeEach(() => {
        store = mockStore({ 
            searchRobots:{
                searchField:"",
            },       
            requestRobots:{
                isPending:false,
                robots:[],
                error:""
            }  
                
            
    });

    store.dispatch = jest.fn();

        component = renderer.create(
            <Provider store={store}>
                <App/>
            </Provider>
        )
    });
    
    it("connection to App container", () => {
        expect (component.toJSON()).toMatchSnapshot();
    })
})



// //Andrei Solution


// describe('App container',()=>{
//     it('expect to render App component', () => {
//         const mockStore = {
//             robots:[],
//             searchField: ''
//         }
//         expect(shallow(<App store={mockStore}/>)).toMatchSnapshot()
//     })
// })



