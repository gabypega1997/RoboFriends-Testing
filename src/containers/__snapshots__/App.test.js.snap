// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Conection React-Redux connection to App container 1`] = `
<div
  className="tc"
>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
  <div
    className="pa2"
  >
    <div
      className="ma3 pa2"
    >
      <button
        id="counter"
        onClick={[Function]}
      >
        Count: 
        0
      </button>
    </div>
    <input
      aria-label="Search Robots"
      className="pa3 ba b--green bg-lightest-blue"
      onChange={[Function]}
      placeholder="search robots"
      type="search"
    />
  </div>
  <div
    style={
      Object {
        "border": "5px solid black",
        "height": "800px",
        "overflow": "scroll",
      }
    }
  >
    <div />
  </div>
</div>
`;
